### PR TITLE
feat: adding course review metrics to content metadata and algolia index

### PR DESCRIPTION
**Pull request by alex-sheehan-edx**
## Ticket Link

Link to the associated ticket
https://2u-internal.atlassian.net/browse/ENT-8126

## Post-review

* [ ] Squash commits into discrete sets of changes
* [ ] Ensure that once the changes have been [deployed to stage](https://gocd.tools.edx.org/go/pipeline/activity/stage-enterprise_catalog), prod is manually deployed

### CODE REVIEW

**Review by kiram15 — APPROVED** [review]
LGTM once tests pass! 🚢

File: enterprise_catalog/apps/api_client/tests/test_discovery.py [hunk 1]
```diff
@@ -116,3 +119,82 @@ def test_get_courses_with_error(self, mock_oauth_client):
...
+        Test _retrieve_course_reviews method with failed response.
+        """
+        mock_oauth_client.return_value.get.side_effect = requests.exceptions.RequestException('error')
+
+        request_params = {'page': 1}
+        client = DiscoveryApiClient()
+        with self.assertRaises(requests.exceptions.RequestException):
+            client._retrieve_course_reviews(request_params)
+
+        assert 'Retrieving course reviews from course-discovery for page 1...' in mock_logger.info.call_args.args[0]
```
Comment (johnnagro) [inline enterprise_catalog/apps/api_client/tests/test_discovery.py]: asserting log lines is less preferable than asserting the mocks are called with a specific input or some outcome/side-effect is observed

File: enterprise_catalog/apps/api_client/discovery.py [hunk 2]
```diff
@@ -80,6 +81,75 @@ def _retrieve_metadata_for_content_filter(self, content_filter, page, request_pa
                 break
         return response.json()
 
+    def _retrieve_course_reviews(self, request_params):
+        """
+        Makes a request to discovery's /api/v1/course_review/ paginated endpoint
```
Comment (johnnagro) [inline enterprise_catalog/apps/api_client/discovery.py:86]: is this endpoint paginated?

File: enterprise_catalog/apps/api/tasks.py [hunk 3]
```diff
@@ -384,15 +387,22 @@ def _update_full_content_metadata_course(content_keys):
...
                 course_run_uuid = json_meta.get('advertised_course_run_uuid')
-                for run in json_meta.get('course_runs'):
+                for run in json_meta.get('course_runs', []):
                     if run.get('uuid') == course_run_uuid:
                         run.update({'start': start_date, 'end': end_date})
 
             # Perform more steps to normalize and move keys around for more consistency across content types.
             _normalize_course_metadata(metadata_record)
 
+            if review := reviews_for_courses_dict.get(content_key):
```
Comment (johnnagro) [inline enterprise_catalog/apps/api/tasks.py:397]: `reviews_for_courses_dict` is a list?

**Review by johnnagro — CHANGES_REQUESTED** [review]

File: enterprise_catalog/apps/api_client/discovery.py (hunk 2 above)
Comment (alex-sheehan-edx) [reply enterprise_catalog/apps/api_client/discovery.py:86]: locally testing it seems so?

File: enterprise_catalog/apps/api/tasks.py (hunk 3 above)
Comment (alex-sheehan-edx) [reply enterprise_catalog/apps/api/tasks.py:397]: nope- `get_course_reviews` should return a dictionary

File: enterprise_catalog/apps/api_client/tests/test_discovery.py (hunk 1 above)
Comment (alex-sheehan-edx) [reply enterprise_catalog/apps/api_client/tests/test_discovery.py]: Removed for a mock called assertion

**Review by johnnagro — APPROVED** [review]

**Pull request merged**